private/bufpkg/bufmodule/uploader.go

```diff
@@ -151,6 +169,11 @@ func (u *uploadOptions) validate() error {
 	if len(u.labels) > 0 && len(u.tags) > 0 {
 		return syserror.New("cannot set both labels and tags")
 	}
+	if u.sourceControlURL != "" {
+		if _, err := url.Parse(u.sourceControlURL); err != nil {
+			return fmt.Errorf("must set a valid url for the source control url: %w", err)
```

Comment:
Do we want to include the URL in the error message or does the error already include it?

i.e. `return fmt.Errorf("invalid source control URL %q: %w", u.sourceControlURL, err)`?

Reply:
The error returned from `net/url` contains the URL already.